File: src/main/java/org/jukeboxmc/event/player/PlayerKickEvent.java
package org.jukeboxmc.event.player;

import org.jukeboxmc.event.Cancellable;
import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class PlayerKickEvent extends PlayerEvent implements Cancellable {

    private String reason;

    /**
     * Creates a new {@link PlayerEvent}
     *
     * @param player who represents the player which comes with this event
     */
    public PlayerKickEvent( Player player, String reason ) {
        super( player );
        this.reason = reason;
    }

    public String getReason() {
        return this.reason;
    }

    public void setReason( String reason ) {
        this.reason = reason;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/player/PlayerKickEvent.kt
package org.jukeboxmc.api.event.player

import org.jukeboxmc.api.event.Cancellable
import org.jukeboxmc.api.player.Player

class PlayerKickEvent(
        player: Player,
        private var reason: String
) : PlayerEvent(player), Cancellable {

    fun getReason(): String {
        return this.reason
    }

    fun setReason(reason: String) {
        this.reason = reason
    }

}